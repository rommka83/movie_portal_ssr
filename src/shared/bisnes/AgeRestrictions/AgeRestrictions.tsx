import React, { HTMLAttributes } from 'react';
import styles from './agerestrictions.module.css';
import classNames from 'classnames';

interface IProps {
  age: number | null;
}

type props = HTMLAttributes<HTMLDivElement> & IProps;

export const AgeRestrictions = React.memo<props>(({ age, className }) => {
  return (
    <div className={classNames(styles.wrapper, className)}>
      <span className={styles.text}>{age ? age : 0} +</span>
    </div>
  );
});
