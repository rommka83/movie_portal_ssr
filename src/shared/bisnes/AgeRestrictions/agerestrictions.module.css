.wrapper {
  width: fit-content;
  padding: 5px;
  background-color: var(--gray-88);
  border-radius: 6px;
}
.text {
  font-weight: 600;
}
