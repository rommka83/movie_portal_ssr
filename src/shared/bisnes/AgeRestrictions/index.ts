export { AgeRestrictions as default } from './AgeRestrictions';
