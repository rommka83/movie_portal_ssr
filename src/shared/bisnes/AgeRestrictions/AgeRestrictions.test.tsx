import renderer from 'react-test-renderer';
import { AgeRestrictions } from './AgeRestrictions';

describe('проверка AgeRestrictions', () => {
  test('отрисовка AgeRestrictions ', () => {
    const component = renderer.create(<AgeRestrictions age={22} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
