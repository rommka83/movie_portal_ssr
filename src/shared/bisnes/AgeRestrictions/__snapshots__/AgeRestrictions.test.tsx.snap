// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка AgeRestrictions отрисовка AgeRestrictions  1`] = `
<div
  className="wrapper"
>
  <span
    className="text"
  >
    22
     +
  </span>
</div>
`;
